### PR TITLE
Added new view to select types for objects

### PR DESCRIPTION
**Pull request by ehconitin**
Issue #6496 
Hi team,

Is this the right approach for handling type selection with states and conditional rendering, or should these be managed on separate pages altogether? Please let me know Ill make changes accordingly :)

I’m also working on styling the buttons according to the Figma design and will be moving constants like categoryDescriptions and categories to the constants folder.

Thanks for your guidance!

https://github.com/user-attachments/assets/452bea9f-4d0a-4472-9941-421b54cda47f

### CODE REVIEW

File: packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategoriesDescription.ts
Comment (ehconitin) [inline packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategoriesDescription.ts:1]: what should be this file named?

File: packages/twenty-front/src/pages/settings/data-model/utils/getFieldTypeOptionsAndConfigs.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {
+    SETTINGS_FIELD_TYPE_CONFIGS,
+    SettingsFieldTypeConfig,
+} from '@/settings/data-model/constants/SettingsFieldTypeConfigs';
+import { SettingsSupportedFieldType } from '@/settings/data-model/types/SettingsSupportedFieldType';
+import { SelectOption } from '@/ui/input/components/Select';
+import omit from 'lodash.omit';
+
+export type ExtendedSelectOption<Value extends string | number | null> =
```
Comment (lucasbordeau) [inline packages/twenty-front/src/pages/settings/data-model/utils/getFieldTypeOptionsAndConfigs.ts:9]: FieldSettingsSelectOption ?

File: packages/twenty-front/src/pages/settings/data-model/SettingsObjectNewField/SettingsObjectNewFieldTypeSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+const StyledButtonContainer = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  gap: 8px;
+  justify-content: start;
+`;
+
+const SettingsObjectNewFieldTypeSelector = ({
+  fieldTypeOptions,
+  handleFieldTypeSelect,
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/pages/settings/data-model/SettingsObjectNewField/SettingsObjectNewFieldTypeSelector.tsx]: done

File: packages/twenty-front/src/pages/settings/data-model/utils/getFieldTypeOptionsAndConfigs.ts (hunk 1 above)
Comment (ehconitin) [reply packages/twenty-front/src/pages/settings/data-model/utils/getFieldTypeOptionsAndConfigs.ts]: done

File: packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategories.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import { CategoryType } from '@/settings/data-model/constants/SettingsFieldTypeConfigs';
+
+export const CATEGORIES: CategoryType[] = ['Basic', 'Relation', 'Advanced'];
```
[the remark this answers is not in the document]
Comment (ehconitin) [reply packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategories.ts]: done

**Comment by ehconitin** [comment]
@lucasbordeau,

I've made the changes as reviewed. However, I need some clarification regarding the blue icons from the Figma design. Is it acceptable if the icons aren't perfectly aligned with the Figma design, as they appear slightly tilted here?

Thanks!

image

File: packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategories.ts
[the remark this answers is not in the document]
Comment (lucasbordeau) [reply packages/twenty-front/src/pages/settings/data-model/constants/SettingsObjectNewFieldTypeCategories.ts:1]: SettingsFieldsTypeCategories same as the constant in PascalCase

File: packages/twenty-front/src/pages/settings/data-model/types/SettingsObjectNewFieldTypeCategories.ts
Comment (lucasbordeau) [inline packages/twenty-front/src/pages/settings/data-model/types/SettingsObjectNewFieldTypeCategories.ts:1]: Same name as the type

**Comment by lucasbordeau** [comment]
> I've made the changes as reviewed. However, I need some clarification regarding the blue icons from the Figma design. Is it acceptable if the icons aren't perfectly aligned with the Figma design, as they appear slightly tilted here?

Let's implement that in another issue !

**Comment by lucasbordeau** [comment]
Could cancel button and browser back button take us to the field selection previous page instead of the field table page ?

**Comment by lucasbordeau** [comment]
Nice ! Let's remove the dropdown because it's redundant now, and we'll be good.

Capture d’écran 2024-08-23 à 16 41 59

**Comment by ehconitin** [comment]
i went through figma to understand on how this page would look after its removal, and every one of them had this dropdown in them.
Also if we remove this dropdown how would the `SettingsObjectFieldEdit` page would look like?
It doesnt really need dropdown because its disabled but its the only way to show user what type is this field of.

**Comment by lucasbordeau** [comment]
@Bonapara Could you decide how we should tell the user which field type he's editing ? 

And also confirm that we don't want to keep the dropdown as it seems redundant with the previous type selection screen ?

**Comment by Bonapara** [comment]
@ehconitin, thanks for this PR!

Do you think you can add the icons and illustrations?

CleanShot 2024-08-27 at 16 26 29

I believe we can avoid recalling the field type in the second step for a V1. We can remove the selection dropdown since there's a preview.

However, we should add a way to navigate back to the first step of the wizard via the breadcrumb. Clicking on the wizard step would open a dropdown menu to select the desired step. Additionally, clicking "New field" should redirect to step 1.

image

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=38395-101881&t=Smks8xwc7zRhJnei-11

Lastly, can you separate the field description into an independent section at the bottom of the page? This way, users can view the preview directly without scrolling.

0bdd24cfc990e2e590916d798f4708d315a7163be34d59f3bd33ae07b5d36649

https://www.figma.com/design/xt8O9mFeLl46C5InWwoMrN/Twenty?node-id=38261-61395&t=Smks8xwc7zRhJnei-11

**Comment by ehconitin** [comment]
Welcome back Thomas!
Those icons and illustrations are to be added in another issue :(
Ill work on the breadcrumb, and separating description field
What about when a user is editing a existing field?
Now we are showing them the field type in the dropdown box, with it (dropdownbox) being disabled.
If we are to remove that dropdown, how to let user know what field type they are editing?

image

**Comment by Bonapara** [comment]
Thanks @ehconitin! Good observation. If a user is editing a field, the field name should appear in the breadcrumb. We might add the field type back in a future PR, but I think we can try it without it for now since there is a preview and you cannot change the field type in edit mode.

**Comment by ehconitin** [comment]
does reopen work?

**Comment by ehconitin** [comment]
@lucasbordeau lost the changes while reverting the main merge this morning
Had to recode whole thing, its almost done, will let yk.

**Comment by ehconitin** [comment]
@Bonapara @lucasbordeau Done! please review.
For now Breadcrumb can not take elements as children we can only provide strings and links. :(

https://github.com/user-attachments/assets/ac7bfa6a-577a-4f67-98a7-e6ce321e987b

**Comment by Bonapara** [comment]
Thanks for your great work @ehconitin 😍

A few feedbacks:

1. I think we have to tackle the breadcrumb in this PR, can you handle it?
2. Can you change the "Relation" description from  "Custom relationships and predefined relationships" to "Create a relation with another object"
3. can you put the card background to `Background/Secondary` and their border-radius to `8px`?

**Comment by ehconitin** [comment]
@Bonapara 
which card ? the values one?
breadcrumb is done!

**Comment by Bonapara** [comment]
Amazing, thanks @ehconitin! 

Can you put the first part of the breadcrumb in tertiary, and add a chevron down light button icon after the step to indicate the step is clickable?

Also in the opened menu can you indicate the option that is currently selected by applying a "Select" state to it?

CleanShot 2024-08-29 at 13 04 54

For the cards, I was talking about these:

CleanShot 2024-08-29 at 13 06 07

**Comment by ehconitin** [comment]
@Bonapara :)

https://github.com/user-attachments/assets/5259bf3c-5f81-435f-a354-3c9557c78ad1

**Comment by ehconitin** [comment]
@lucasbordeau 
Ready for review!

**Review by lucasbordeau — APPROVED** [review]

**Review by lucasbordeau — APPROVED** [review]

**Comment by Bonapara** [comment]
Hi @ehconitin! Thanks again for this. The field type cards' background should be `Background/Secondary` (instead of white) if still possible to make the fix.

**Comment by ehconitin** [comment]
@Bonapara 
I did gave it as secondary, i think the issue is `SettingsPageContainer` having background as secondary which is making this buttons blend in.
The SettingsPageContainer should have primary as background.
i think its the same issue in #6791 
Charles suggested the same [comment](https://github.com/twentyhq/twenty/pull/6794#issuecomment-2320516226) 
Should I make changes in this PR or it will be done in that issues PR? Its a one line change.

**Comment by charlesBochet** [comment]
@ehconitin Wow, that's impressive work

I'm taking a look

**Comment by ehconitin** [comment]
@charlesBochet any corrections/changes required?
please let me know!

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**